contentService/src/main/java/com/kovan/app/controller/BucketController.java

```diff
@@ -0,0 +1,49 @@
+package com.kovan.app.controller;
+
+import com.kovan.app.service.BucketService;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import java.io.IOException;
+
+@RestController
+@RequestMapping("/bucket")
+public class BucketController {
+
+   private final BucketService bucketService;
+
+    public BucketController(BucketService bucketService) {
+        this.bucketService = bucketService;
+    }
+
+    @GetMapping("/download")
+    public ResponseEntity<Resource> downloadFile(@RequestParam("fileName") String fileName) throws IOException {
```

Comment:
user won't provide file name user will give some id you try to get the meta data from db and do the process

Reply:
getting id from user.